### PR TITLE
CP-38759: Add pool.deconfigure_repository_proxy

### PR DESCRIPTION
**Pull request by minglumlu**

### CODE REVIEW

File: ocaml/idl/datamodel_pool.ml [hunk 1]
```diff
@@ -803,6 +803,14 @@ open Datamodel_types
       ~allowed_roles:_R_POOL_OP
       ()
 
+  let deconfigure_repository_proxy = call
```
Comment (lindig) [inline ocaml/idl/datamodel_pool.ml:806]: I think "deconfigure" is a quite unusual verb. I would have preferred any of "decommission", "disable", "remove". Another option is "clear", which has a different meaning.

File: ocaml/idl/datamodel_pool.ml [hunk 2]
```diff
@@ -803,6 +803,14 @@ open Datamodel_types
...
+      ~name:"deconfigure_repository_proxy"
+      ~in_product_since:rel_next
+      ~doc:"Clear the configured proxy for RPM package repositories."
```
Comment (lindig) [inline ocaml/idl/datamodel_pool.ml:809]: "Clear" is not the same as "deconfigure" - it reads like a contradiction. Clearing a cache means to reset it to an empty state whereas deconfigure means to remove it (in my understanding).

File: ocaml/xapi-cli-server/cli_frontend.ml [hunk 3]
```diff
@@ -633,6 +633,15 @@ let rec cmdtable_data : (string * cmd_spec) list =
       ; flags= []
       }
     )
+  ; ( "pool-deconfigure-repository-proxy"
+    , {
+        reqd= []
+      ; optn= []
+      ; help= "Clear configured proxy for RPM package repositories"
```
Comment (lindig) [inline ocaml/xapi-cli-server/cli_frontend.ml:640]: See above - what this does is not clear enough.

File: ocaml/idl/datamodel_pool.ml (hunk 1 above)
Comment (minglumlu) [reply ocaml/idl/datamodel_pool.ml]: Thanks. I prefer to use "remove" in this case.

Comment (minglumlu) [reply ocaml/idl/datamodel_pool.ml]: But the SDK has constraint on the usage of "remove". So I change to use "reset".

Comment (lindig) [reply ocaml/idl/datamodel_pool.ml]: The updated code uses "reset", though. So maybe we should get an idea what this code is supposed to do first. What happens when this API call is executed and how does it affect future API calls?

Comment (minglumlu) [reply ocaml/idl/datamodel_pool.ml]: The call is just to remove the configured proxy, or in other words, change it to the initial default value which stands for "no proxy" configured.

Comment (lindig) [reply ocaml/idl/datamodel_pool.ml]: I think "remove" would be most appropriate, followed by "disable". "Reset" and "clear" would suggest the cache still exists, is initially empty, but would fill up again. If the cache does not fill up again, it is disabled.

Comment (robhoes) [reply ocaml/idl/datamodel_pool.ml]: It's not actually a cache, but a network proxy. "Reset" to me means that it's not necessarily going away. Perhaps "disable" is the best if "remove" isn't possible.

**Review by lindig — APPROVED** [review]

**Review by robhoes — APPROVED** [review]

**Pull request merged**